src/detect-ssl-version.c

```diff
@@ -239,6 +261,10 @@ static DetectSslVersionData *DetectSslVersionParse(const char *str)
                 ssl->data[TLS12].ver = TLS_VERSION_12;
                 if (neg == 1)
                     ssl->data[TLS12].flags |= DETECT_SSL_VERSION_NEGATED;
+            } else if (strncasecmp("tls1.3", tmp_str, 6) == 0) {
```

Comment:
strNcasecmp seems wrong here, it would match "tls1.333333" as well. Could you update them all?

Reply:
Would checking the string length be a sane thing to do instead?

Reply:
These are all nul terminated strings, so it shouldn't be necessary to explicitly check the length.

Reply:
My mind was thinking of strstr not on strcmp (which does exact match), so yeah, using strncasecmp does not make sense at all! I'll change to strcasecmp.